Lib/blackrenderer/backends/sweepGradient.py

```diff
@@ -0,0 +1,58 @@
+from math import pi, ceil, sin, cos
+from fontTools.misc.vector import Vector
+
+
+def buildSweepGradientPatches(
+    colorLine,
+    center,
+    radius,
+    startAngle,
+    endAngle,
+    useGouraudShading,
+):
+    """Provides colorful patches that mimic a sweep gradient, for use, in
+    particular, in the Cairo and CoreGraphics backends, since these libraries
+    lack the sweep gradient feature."""
+    patches = []
+    # generate a fan of 'triangular' bezier patches, with center 'center' and radius 'radius'
+    degToRad = pi / 180.0
+    if useGouraudShading:
+        maxAngle = pi / 360.0
+        radius = 1.05 * radius  # we will use straight-edged triangles
```

Comment:
Could this perhaps be the responsiility of the caller?

Reply:
Or we should comment that we expect the resulting triangles/patches to be clipped, and guarantee that the radius is completely filled, but may go over. I'm thinking we may need to do similar things for the PAD extend mode: it's perhaps easiest to draw padding triangle that is too big.

Reply:
I replaced 1.05 by the correct value (1 / cos(maxAngle/2)) so that the straight triangle always cover the disk with the original radius (which is expected to cover the clip-path or whatever needs to be painted, and is responsibility of the caller)

Reply:
Great, thanks!